src/pages/volunteer.js

```diff
@@ -0,0 +1,149 @@
+import FormField from "../components/FormField";
+
+const OfferHelpPage = () => {
+    const router = useRouter();
+    const [isSubmitting, setIsSubmitting] = useState(false);
+    const [serverError, setServerError] = useState("");
+    const { register, handleSubmit, errors } = useForm({
+        mode: "onBlur"
+    });
+
+    const onSubmit = async data => {
+        setIsSubmitting(true);
+        setServerError("");
+        try {
+            const volunteer = {
+                phone: data.phone,
+                name: data.name,
+                email: data.email,
+                zipcode: String(data.zipcode)
+            };
+
+            await axios.post(
+                "https://us-central1-neighbor-army.cloudfunctions.net/widgets/worker",
+                volunteer
+            );
+
+            router.push("/volunteer-success");
+        } catch (err) {
+            setIsSubmitting(false);
+            if (err.response && err.response.status === 500) {
+                setServerError(
+                    "There is already a volunteer with this phone number"
+                );
+            }
+        }
+    };
+
+    return (
+        <main>
+            <header>
+                <h1>Sign up to be a Volunteer</h1>
+                <p>We appreciate your interest. How may we get in touch?</p>
+            </header>
+            {serverError && (
+                <section className="server-error">
+                    <h2>{serverError}</h2>
+                </section>
+            )}
+            <form
+                className="offer-help__form"
+                onSubmit={handleSubmit(onSubmit)}
+            >
+                <FormField
+                    labelText="Full Name"
+                    errorMsg={errors.name && "Please enter your full name"}
+                    placeholder="e.g. Jane Smith"
+                    name="name"
+                    ref={register({
+                        required: true
+                    })}
+                />
+                <FormField
+                    labelText="Email Address"
+                    errorMsg={errors.email && "Please enter your email"}
+                    placeholder="e.g. jane.smith@gmail.com"
+                    name="email"
+                    type="email"
+                    ref={register({
+                        required: true,
+                        pattern: /^\S+@\S+\.\S+$/
+                    })}
+                />
+                <FormField
+                    labelText="Phone Number"
+                    errorMsg={errors.phone && "Please enter your phone number"}
+                    placeholder="e.g. (425) 555-2323"
+                    name="phone"
+                    type="tel"
+                    ref={ref =>
+                        ref &&
+                        register(ref.inputElement, {
+                            required: true,
+                            pattern: /^(\+\d{1,2}\s)?\(?\d{3}\)?[\s.-]?\d{3}[\s.-]?\d{4}$/
```

Comment:
Fair point. We should change this in the future, when we start targeting places outside the USA. But, right now, the entire infrastructure is dependent on USA ZipCodes, so making it flexible towards the rest of the world is a feature for the roadmap. 

In any case, I didn't write that code (just refactored it) so if it was a bug, it's one that exists in production right now. But it's not. But it could be.

Reply:
I understand